### PR TITLE
Comm fix

### PR DESCRIPTION
**Pull request by pawelkaczmarekuk**
### Change description ###
fixes:
- pop up rotation
- clipped comment text
- comment right side alignment

### CODE REVIEW

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/rectangle/popup-toolbar/popup-toolbar.component.ts [hunk 1]
```diff
@@ -10,7 +11,38 @@ export class PopupToolbarComponent {
   readonly HEIGHT = 70;
   readonly WIDTH = 350;
 
-  @Input() top: number;
-  @Input() left: number;
+  @Input() rectangle: Rectangle;
+  @Input() zoom = 1;
+  @Input() rotate = 0;
+
+  popupStyles() {
```
Comment (linusnorton) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation/rectangle/popup-toolbar/popup-toolbar.component.ts]: Is there some way we could wrap the commonality between this and the other rotate code into a service?

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/comment/comment.component.ts [hunk 2]
```diff
@@ -102,4 +103,9 @@ export class CommentComponent {
     this.updated.emit(this.fullComment);
     this.editable = false;
   }
+
+  private getFirstNonNullParentProperty(el: Node, property: string) {
+    return el.parentNode ? ( el.parentNode[property] ?
```
Comment (linusnorton) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation/comment/comment.component.ts]: I'm not sure if this is better....

`return !el.parentNode ? null : el.parentNode[property] || this.getFirstNonNullParentProperty(el.parentNode, property);`

File: projects/media-viewer/src/lib/viewers/pdf-viewer/pdf-js/pdf-js-wrapper.ts [hunk 3]
```diff
@@ -116,10 +116,14 @@ export class PdfJsWrapper {
     return this.pdfViewer.pagesRotation = (this.pdfViewer.pagesRotation + rotation) % 360;
   }
 
-  public getRotate(): number {
+  public getPagesRotation(): number {
     return this.pdfViewer.pagesRotation;
   }
 
+  public getNormalisedPagesRotation(): number {
```
Comment (linusnorton) [inline projects/media-viewer/src/lib/viewers/pdf-viewer/pdf-js/pdf-js-wrapper.ts]: why have both of these methods, surely we just need one

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/rectangle/popup-toolbar/popup-toolbar.component.ts (hunk 1 above)
Comment (pawelkaczmarekuk) [reply projects/media-viewer/src/lib/annotations/annotation-set/annotation/rectangle/popup-toolbar/popup-toolbar.component.ts]: Nothing straight forward comes to my mind. We could refactor later?

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/comment/comment.component.ts (hunk 2 above)
Comment (pawelkaczmarekuk) [reply projects/media-viewer/src/lib/annotations/annotation-set/annotation/comment/comment.component.ts]: OK

File: projects/media-viewer/src/lib/viewers/pdf-viewer/pdf-js/pdf-js-wrapper.ts (hunk 3 above)
Comment (pawelkaczmarekuk) [reply projects/media-viewer/src/lib/viewers/pdf-viewer/pdf-js/pdf-js-wrapper.ts]: I'll remove the getPagesRotation

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/comment/comment.component.ts (hunk 2 above)
Comment (pawelkaczmarekuk) [reply projects/media-viewer/src/lib/annotations/annotation-set/annotation/comment/comment.component.ts]: ok

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/comment/comment.component.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-  <form (click)="onCommentClick()" class="aui-comment"
+  <form #form (click)="onCommentClick()" class="aui-comment"
```
Comment (linusnorton) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation/comment/comment.component.html]: what is the #form for?

Comment (pawelkaczmarekuk) [reply projects/media-viewer/src/lib/annotations/annotation-set/annotation/comment/comment.component.html]: to grab native element and its parent to eventually get clientWidth

**Pull request merged**